src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/ShutdownTask.java

```diff
@@ -93,24 +101,42 @@ public TaskResult call() {
         boolean applicationException = false;
 
         try {
+            ShutdownReason localReason = reason;
+            List<Shard> allShards = new ArrayList<>();
```

Comment:
We allow this variable to be null. onShardConsumerShutDown can take null list of shards. If null list of shards is passed in, it will call listShards again in the syncer